cmake/linker_script/common/common-rom.cmake

```diff
@@ -156,6 +156,10 @@ if(CONFIG_PCIE)
   zephyr_linker_section_configure(SECTION irq_alloc INPUT ".irq_alloc*" KEEP SORT NAME)
 endif()
 
+if(CONFIG_NEWLIB_LIBC_HEAP_LISTENER)
+  zephyr_iterable_section(NAME z_libc_heap_listener KVMA RAM_REGION GROUP RODATA_REGION SUBALIGN 4)
+endif()
```

Comment:
But currently Zephyr only provides `_sbrk` for newlib. Should I make the Kconfig generic even if the listener will not be used for other libc implementations for now?

Reply:
Since you are laying out a general (not newlib-specific) framework for the libc heap size change notifications here, it would only make sense to add a general Kconfig that enables it, which each libc-specific config can select.

Without `CONFIG_LIBC_HEAP_LISTENER`, we will likely end up with something like this later:
```
if(CONFIG_NEWLIB_LIBC_HEAP_LISTENER or CONFIG_MWDT_LIBC_HEAP_LISTENER)
...
```

With `CONFIG_LIBC_HEAP_LISTENER`, libc-specific configs only need to select `LIBC_HEAP_LISTENER` instead of adding `or ...` to everywhere `CONFIG_NEWLIB_LIBC_HEAP_LISTENER` is referenced:
```
config MWDT_LIBC_HEAP_LISTENER
...
	select LIBC_HEAP_LISTENER
...
```

As you can see, the former is much cleaner.

Reply:
Ok, makes sense, thanks for explanation.

Reply:
Done